packages/cli/src/commands/generate/service/templates/test.ts.template

```diff
@@ -1,28 +1,39 @@
-<% // To JSON stringify and remove quotes around `scenario` variables,
-   // if `asDate` is true, it will return dates as new Date(...) declarations,
-   // ex: `new Date("2022-09-01T00:21:58.876Z")
-   // otherwise, dates will be returned as strings. ex: "2022-09-01T00:21:58.876Z"
-const stringifyValue = (obj, asDate=false) => {
-  return JSON.stringify(obj).replace(/['"].*?['"]/g, (string) => {
-    if (string.match(/scenario\./)) {
-      return string.replace(/['"]/g, '')
-    }
-
-    // BigInt
-    if (string.match(/^\"\d+n\"$/)) {
-      return string.substr(1, string.length - 2)
-    }
-
-    if(asDate){
-      // Creates a new Date declaration if the string matches this ISO8601 date format.
-      // ex: "2022-09-01T00:21:58.876Z" will return `new Date("2022-09-01T00:21:58.876Z")`
-      if (string.match(/^\"\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}Z\"$/)) {
-        return `new Date(${string})`
+<% // Extracts the values of an object by converting the JSON into a string.
+   // Looks for quoted strings, either by single (') ro double (") quotes.
+   // When found
+   // - Removes the quotes around `scenario` variables.
+   // - Removes the quotes around `BigInt` fields.
+   // It can also return object instances via the `options` parameter
+   // - `options.useDateObjects` default value: `false`
+   //   - false: will return date values as a string. ex: "2022-09-01T00:21:58.876Z"
+   //   - true: will return date values as a date. ex: `new Date("2022-09-01T00:21:58.876Z")`
+  const extractValues = (obj, options={ useDateObjects: false }) => {
+    return JSON.stringify(obj).replace(/['"].*?['"]/g, (string) => {
+      if (string.match(/scenario\./)) {
+        return string.replace(/['"]/g, '')
       }
-    }
-    return string
-  })
-} %>
+
+      // BigInt
+      if (string.match(/^\"\d+n\"$/)) {
+        return string.substr(1, string.length - 2)
+      }
+
+      if (options.useDateObjects) {
+        // Creates a new Date declaration if the string matches this ISO8601 date format.
+        // ex: "2022-09-01T00:21:58.876Z" will return `new Date("2022-09-01T00:21:58.876Z")`
+        if (string.match(/^\"\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}Z\"$/)) {
+          return `new Date(${string})`
+        }
+      }
+
+      return string
+    })
+  }
+
+  // This is an overload of extractValues
```

Comment:
Technically this isn't an overload. Overloaded functions have the same name, just different number of arguments. 

My suggestion is to just remove the current text and instead replace it with an explanation of what this function is used for, and why it's needed.

Reply:
done